### PR TITLE
Context2

### PR DESCRIPTION
**Pull request by MarikaBBB**
### Summary

This pull request brings several refactors and improvements to make the code easier to read, more organised, and faster in handling cart functions, button components, and context usage.

**CartSummary Component**:
  - Removed extra calculations for `finalTotal` and simplified conditionals, making the code clearer and more efficient.

**Quantifier and QuantifierBtn Components**:
  - Moved `QuantifierBtnClasses` to a constant, avoiding recreation on each render to boost performance.
  - Optimised class names to make styling simpler and more consistent.

**Button Component**:
  - Simplified prop handling, making it easier to reuse with an improved `className` setup that’s more flexible across UI elements.

**Cart Context and Reducer**:
  - Added action constants (`ADD_TO_CART`, `DECREASE_QUANTITY`) to improve readability and avoid hard-coded action types.
  - Improved localStorage sync to keep cart data consistent across sessions.
  - Optimised state calculation, making context more reliable and easier to maintain.

**ProductCard Component**:
  - Simplified prop destructuring for a cleaner structure.
  - Integrated context alignment for smoother dispatch actions, ensuring `ADD_TO_CART` works seamlessly. 
 
**FooterSection Component**:
- Defined explicit types for props (`title`, `links`, `isHorizontal`) to resolve TypeScript errors.
- Updated `Link` interface to accept JSX elements for `name`, enabling the display of social media icons in the footer.

**Socials Component**:
- Updated social media links to pass JSX elements as `name` for social icons.

**Additional Improvements**:
- Addressed multiple TypeScript errors by defining explicit types and updating interfaces, improving type safety and compatibility.
- Simplified and organised components to promote reusability, particularly in the cart and footer sections.
- - Removed empty files

### CODE REVIEW

File: src/Context/Cart.tsx [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
   totalPrice: number;
 }
 
-// Define the shape of the actions
+
 interface CartAction {
-  type: "ADD_TO_CART" | "REMOVE_FROM_CART" | "DECREASE_QUANTITY";
+  type: typeof ADD_TO_CART | typeof DECREASE_QUANTITY;
   payload: Vinyl;
 }
```
Comment (JoshCodedit) [inline src/Context/Cart.tsx:18]: Could this interface be moved to the types fil. And then you can import it similar to how you have done with the vinyl types

import { CartAction } from "../utils/types";

File: src/footer-components/FooterSection.tsx [hunk 2]
```diff
@@ -1,6 +1,16 @@
+interface Link {
+  url: string;
+  name: React.ReactNode;
+}
 
+interface FooterSectionProps {
+  title: string;
+  links: Link[];
+  isHorizontal: boolean;
+}
```
Comment (JoshCodedit) [inline src/footer-components/FooterSection.tsx:11]: These interfaces could be added to the types file too

File: src/footer-components/FooterSection.tsx [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
 
-const FooterSection = ({ title, links, isHorizontal }) => {
+
+const FooterSection: React.FC<FooterSectionProps> = ({ title, links, isHorizontal }) => {
```
Comment (JoshCodedit) [inline src/footer-components/FooterSection.tsx]:
```suggestion
const FooterSection = ({ title, links, isHorizontal }: FooterSectionProps) => {
```

**Review by JoshCodedit — COMMENTED** [review]
This looks much more cleaner. Nice work Marika 👍👍

Just a couple of suggestions

File: src/Context/Cart.tsx (hunk 1 above)
Comment (MarikaBBB) [reply src/Context/Cart.tsx:18]: Thank Josh. I moved it to the type.ts. By doing this I have to twik some costants.

File: src/footer-components/FooterSection.tsx (hunk 2 above)
Comment (MarikaBBB) [reply src/footer-components/FooterSection.tsx:11]: done it

**Review by maxitect — APPROVED** [review]
Thanks Marika, really nice work!

**Review by JoshCodedit — APPROVED** [review]

**Pull request merged**